samcli/commands/build/build_context.py

```diff
@@ -265,3 +265,27 @@ def _collect_single_buildable_layer(
             raise MissingBuildMethodException(f"Build method missing in layer {resource_identifier}.")
 
         resource_collector.add_layer(layer)
+
+    @staticmethod
+    def _is_function_buildable(function: Function):
+        # no need to build inline functions
+        if function.inlinecode:
+            LOG.debug("Skip building inline function: %s", function.full_path)
+            return False
+        # no need to build functions that are already packaged as a zip file
+        if isinstance(function.codeuri, str) and function.codeuri.endswith(".zip"):
```

Comment:
Are we going to consider an S3 location as well?

Reply:
I am still checking details why we handled the S3 this way (this code exists before our git history) with @jfuss. I am gonna handle s3 in a separate PR

```py
        codeuri = resource_properties.get(code_property_key, SamBaseProvider.DEFAULT_CODEURI)
        # CodeUri can be a dictionary of S3 Bucket/Key or a S3 URI, neither of which are supported
        if isinstance(codeuri, dict) or (isinstance(codeuri, str) and codeuri.startswith("s3://")):
            if not ignore_code_extraction_warnings:
                LOG.warning(
                    "Lambda function '%s' has specified S3 location for CodeUri which is unsupported. "
                    "Using default value of '%s' instead",
                    name,
                    SamBaseProvider.DEFAULT_CODEURI,
                )
            return SamBaseProvider.DEFAULT_CODEURI
        return cast(str, codeuri)
```